### PR TITLE
Add button style prop to date time picker component

### LINKED ISSUE #1157: Add Button Style Prop to DateTime Picker Component
@praveen-murali-ind _a Ref: https://github.com/bigbinary/neeto-planner-web/issues/1480

### PR DESCRIPTION
**Pull request by Tajwebdeveloper**
Fixes #1157 

**Description**
1. Provided option to add a custom element to be passed in the date and time picker.
2. `customTarget` prop added to `<TimePicker />` and `<DatePicker />` components.
3. we can now pass element like this `<DatePicker customTarget={<Button icon={Calendar} />} {...args} />`

**Checklist**

- [x] I have performed a self-review of my code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added the necessary label (patch/minor/major - If package publish is required)
- [x] I have followed the suggested description format and styling

**Reviewers**

### CODE REVIEW

**Comment by amaldinesh7** [comment]
@Tajwebdeveloper _A

1. Please fix the PR name (Should not include the issue number)
2. Please follow the format specified in the PR template while writing the description. (No image should be added in the description section, it can be added separately as a comment)
3. There are various edge cases to consider while giving an option for a custom target. Refer - https://vimeo.com/725556643

**Comment by Tajwebdeveloper** [comment]
image
image

**Comment by Tajwebdeveloper** [comment]
@amaldinesh7 I've updated the code and fixed the issues. Also added a few more examples to the storybook.

**Comment by karthiknmenon** [comment]
@Tajwebdeveloper Please assign the PR back to the reviewer once the changes have been made. You could use gitEmit and type in `@GitHub-UserName-Of-Reviewer _a` to assign the PR for review.

**Comment by amaldinesh7** [comment]
@ajmaln _a Please take a look at this changes.

File: lib/components/TimePicker.js [hunk 1]
```diff
@@ -41,21 +41,60 @@ const TimePicker = forwardRef(
...
       value,
+      customTarget,
       ...otherProps
     },
     ref
   ) => {
     const id = useId(otherProps.id);
     const timePickerRef = useSyncedRef(ref);
-
+    const [currentValue, setCurrentValue] = useState(defaultValue);
```
Comment (ajmaln) [inline lib/components/TimePicker.js:51]: same here

File: lib/components/TimePicker.js [hunk 2]
```diff
@@ -41,21 +41,60 @@ const TimePicker = forwardRef(
...
+                }}
+                {...otherProps}
+                picker="time"
+                mode={undefined}
+                defaultValue={convertToDayjsObjects(defaultValue)}
+                value={convertToDayjsObjects(value)}
+              />
+            </div>
+          </div>
+        ) : <Component
```
Comment (ajmaln) [inline lib/components/TimePicker.js:97]: Can we extract the logic to be a wrapper component that conditionally adds the `customTarget` if defined and reuse the same in `DatePicker` as well.

**Review by ajmaln — COMMENTED** [review]
@Tajwebdeveloper _a Please check the suggestions and please reformat the files.

One thing I noticed is that the `pointer` cursor is not showing up for the button custom targets instead the text edit cursor is appearing. Please check that as well.

**Comment by sandip-mane** [comment]
Any update here? This is a blocker for one [issue in neetoPlanner](https://github.com/bigbinary/neeto-planner-web/issues/1480) for a long time now.

**Comment by amaldinesh7** [comment]
> Any update here? This is a blocker for one [issue in neetoPlanner](https://github.com/bigbinary/neeto-planner-web/issues/1480) for a long time now.

Will resolve ASAP. The solution implemented in the PR is quite complex, need to find ways to simplify it.

**Comment by amaldinesh7** [comment]
@sandip-mane @Melvin9 @Tajwebdeveloper @karthiknmenon 
Please watch this video https://vimeo.com/734301060/96ad1155d1. I've mentioned 2 workarounds to achieve the use case. 

Examples:
With DatePicker props - https://codesandbox.io/s/polished-cherry-24g69v?file=/index.js
With Dropdown component - https://codesandbox.io/s/interesting-newton-ykjmgq?file=/src/App.js

cc: @ajmaln @praveen-murali-ind

**Comment by sandip-mane** [comment]
@amaldinesh7 I think we should add this example for customizing datepicker components in the neetoUI's documentation for datepicker. What do you think?

**Comment by amaldinesh7** [comment]
> @amaldinesh7 I think we should add this example for customizing datepicker components in the neetoUI's documentation for datepicker. What do you think?

Yes. I will add both the examples.

**Pull request closed without merging**